dialer.go

```diff
@@ -331,25 +332,34 @@ func (d *Dialer) Dial(ctx context.Context, icn string, opts ...DialOption) (conn
 	}), nil
 }
 
-func invalidClientCert(cn instance.ConnName, l debug.Logger, c *tls.Config) bool {
+// validClientCert checks that the ephemeral client certificate retrieved from
+// the cache is unexpired. The time comparisons strip the monotonic clock value
+// to ensure an accurate result, even after laptop sleep.
+func validClientCert(cn instance.ConnName, l debug.Logger, c *tls.Config) bool {
 	// The following conditions should be impossible (no certs, nil leaf), but
 	// just in case there's an unknown edge case, check assumptions before
 	// proceeding.
 	if len(c.Certificates) == 0 {
-		return true
+		return false
 	}
 	if c.Certificates[0].Leaf == nil {
-		return true
+		return false
 	}
-	now := time.Now()
-	notAfter := c.Certificates[0].Leaf.NotAfter
+	// Use UTC() to strip monotonic clock value to guard against inaccurate
+	// comparisons, especially after laptop sleep.
+	// See the comments on the monotonic clock in the Go documentation for
+	// details: https://pkg.go.dev/time#hdr-Monotonic_Clocks
+	now := time.Now().UTC()
+	expiration := c.Certificates[0].Leaf.NotAfter.UTC()
+	valid := expiration.After(now)
 	l.Debugf(
 		"[%v] Now = %v, Current cert expiration = %v",
 		cn.String(),
-		now.UTC().Format(time.RFC3339),
-		notAfter.Format(time.RFC3339),
+		now.Format(time.RFC3339),
+		expiration.Format(time.RFC3339),
 	)
-	return now.After(notAfter)
+	l.Debugf("[%v] Cert is valid = %v", cn.String(), valid)
```

Comment:
does this line add any value? The previous debug line showcases both times so you can already see if the cert is valid, curious if we want this extra debug or if it will clutter.

Reply:
This log line is how I discovered the bug -- the check was reporting "true" even though the cert time was obviously invalid.

Reply:
Let's make sure we update our docs on where we want to add logging across the board then, this may be something we want to log in other libraries as well.